jkube-kit/config/resource/src/test/java/org/eclipse/jkube/kit/config/access/ClusterAccessTest.java

```diff
@@ -13,101 +13,84 @@
  */
 package org.eclipse.jkube.kit.config.access;
 
-import java.net.UnknownHostException;
-import java.util.function.Consumer;
-
-import org.eclipse.jkube.kit.common.KitLogger;
-
-import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.api.model.APIGroupListBuilder;
 import io.fabric8.kubernetes.client.KubernetesClient;
-import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.server.mock.EnableKubernetesMockClient;
+import io.fabric8.kubernetes.client.server.mock.KubernetesMockServer;
 import io.fabric8.openshift.client.OpenShiftClient;
-import org.junit.jupiter.api.AfterEach;
+import org.eclipse.jkube.kit.common.KitLogger;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.mockito.MockedConstruction;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.contains;
 import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.ArgumentMatchers.isNull;
 import static org.mockito.ArgumentMatchers.startsWith;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.mockConstruction;
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
 
+@EnableKubernetesMockClient(https = false)
 class ClusterAccessTest {
 
   private KitLogger logger;
-  private MockedConstruction<DefaultKubernetesClient> kubernetesClientMockedConstruction;
-  private Consumer<OpenShiftClient> onInstantiate;
+  private KubernetesMockServer mockServer;
+  private ClusterConfiguration clusterConfiguration;
 
   @BeforeEach
   void setUp() {
     logger = spy(new KitLogger.SilentLogger());
-    onInstantiate = oc -> {
-    };
-    kubernetesClientMockedConstruction = mockConstruction(DefaultKubernetesClient.class, (mock, ctx) -> {
-      final OpenShiftClient oc = mock(OpenShiftClient.class);
-      when(mock.adapt(OpenShiftClient.class)).thenReturn(oc);
-      onInstantiate.accept(oc);
-    });
-  }
-
-  @AfterEach
-  void tearDown() {
-    kubernetesClientMockedConstruction.close();
+    clusterConfiguration = ClusterConfiguration.builder()
```

Comment:
it may be good to explain in the PR description why these changes were made and any reference to a kubernetes-client PR/issue

Reply:
These changes are just to improve the test and make use of the Kubernetes Mock Server provided by the client instead of relying on complex Mocks.

Why now? The Kubernetes Mock server as become more stable and performant over the last couple of years. Being able to test `ClusterAccess` against a semi-real cluster is easier to maintain than relying on Mocks that need extra-maintenance. In this case, the Client is slowly moving to abstract away explicit KubernetesClient implementations and moving on to SPI provided and more generic ones instead.

Reply:
this is good for me, ideally in the description or in another PR